cli/awsbatch/awsbsub.py

```diff
@@ -249,6 +242,153 @@ def _upload_and_get_command(boto3_factory, args, job_name, region, s3_bucket, lo
     return command
 
 
+def _get_stdin_and_upload(boto3_factory, s3_bucket, job_folder, job_script, timeout):
+    """
+    Create file from STDIN and upload to S3.
+
+    :param boto3_factory: initialized Boto3ClientFactory object
+    :param s3_bucket: S3 bucket
+    :param job_folder: S3 bucket subfolder
+    :param job_script: job script name
+    :param timeout: S3 expiration time in seconds
+    """
+    try:
+        # copy stdin to temporary file
+        with os.fdopen(sys.stdin.fileno(), "rb") as src:
+            with open(job_script, "wb") as dst:
+                shutil.copyfileobj(src, dst)
+
+        _upload_to_s3(boto3_factory, s3_bucket, job_script, job_folder + job_script, timeout)
```

Comment:
I'm totally fine with leaving it as it is but this function could become part of an S3Uploader class that holds the factory and the bucket as part of its attributes so that these don't need to be passed around at every invocation.

Reply:
I totally agree